qiskit/chemistry/algorithms/eigen_solvers/q_eom_vqe.py

```diff
@@ -105,7 +107,7 @@ def _run(self):
         self._quantum_instance.circuit_summary = True
         opt_params = self._ret['opt_params']
         logger.info("opt params:\n%s", opt_params)
-        wave_fn = self._var_form.construct_circuit(opt_params)
+        wave_fn = self.get_optimal_circuit()
```

Comment:
Does the former not work. I would like to remove the result methods off the variational algo so all the result exists only in the result object and the algo does not have to have these stateful methods based on whether it ran or not.

Reply:
The former works but that's exactly what `get_optimal_circuit` does. Not calling this function would mean we need to copy+paste the same there, which is more than 1 line due to the varform-circuit distinction.

I'm not sure I understand the second part of the comment, the variable `opt_params` is only set when the algo is run, equivalently, `get_optimal_circuit` only makes sense when the algo is run 🤔

Reply:
I mean remove the stateful methods from VQAlgorithm so that the results, that come from running, are in the result object and not stored in the class too.